### PR TITLE
chore: create latest immutable state nexus

### LINKED ISSUE #9684: Create latest immutable state nexus
The signed state manager currently holds a reference to the latest immutable state. Since the signed state manager should become a component, it will not have any getter methods by design. So some of the functionality needs to be extracted.

Currently this is used for:
- handling precons tx
- app getter
- dumping the latest immutable state to disk
- pces logging

Does this state need to be hashed for any of these purposes?

Note: the tx handling of this state will be done in the future (#9715)

### PR DESCRIPTION
**Pull request by lpetrovic05**
closes #9684

### CODE REVIEW

**Review by mxtartaglia-sl — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 1]
```diff
@@ -1376,9 +1406,10 @@ public AddressBook getAddressBook() {
     @SuppressWarnings("unchecked")
     @Override
     public <T extends SwirldState> AutoCloseableWrapper<T> getLatestImmutableState(@NonNull final String reason) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:1408]: This method could now be annotated `@NonNull`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventReplayWorkflow.java [hunk 2]
```diff
@@ -72,8 +72,8 @@ public static void replayPreconsensusEvents(
             @NonNull final QueueThread<GossipEvent> intakeQueue,
             @NonNull final ConsensusRoundHandler consensusRoundHandler,
             @NonNull final QueueThread<ReservedSignedState> stateHashSignQueue,
-            @NonNull final StateManagementComponent stateManagementComponent,
-            final long initialMinimumGenerationNonAncient) {
+            final long initialMinimumGenerationNonAncient,
+            @NonNull final Supplier<ReservedSignedState> latestImmutableState) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventReplayWorkflow.java:76]: missing param doc for `latestImmutableState`

**Review by alittley — DISMISSED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java (hunk 1 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:1408]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventReplayWorkflow.java (hunk 2 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PreconsensusEventReplayWorkflow.java:76]: added

**Review by cody-littley — DISMISSED** [review]

**Review by imalygin — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Pull request merged**